### DISCUSSION TITLE
Using declared variables as sent Modbus frames at Scripting's SML

### DISCUSSION THREAD

**Discussion by Erickrk** [arendst/Tasmota#17231] [Q&A] [answered]
Hello everyone, I've had the idea to do the following:

>D
;Declares Modbus frame as a string variable
testF="01040000"

>B
->sensor53 r
>M 1
+1,3,m,0,9600,MODBUS,1,1,**%testF%**,01040002,01040004,01040006,01040008,0104000a,0104000c,0104000e,01040010

Which would be helpful to reduce the script's size, but it doesn't seem to work with this syntax. Is there any other implementation for this?
I've seen the [Resol Detasol example](https://tasmota.github.io/docs/Smart-Meter-Interface/#resol-deltasol-bs-plus-vbus) but I didn't get how the "solar" variable is created.

Looking forward to hearing from you!

**Comment by gemu2015** [comment]
#define SML_REPLACE_VARS
and define a string variable in section >D
however only decoder variables are supported, so not applicable in the + descriptor line.

**Reply by Erickrk** [reply]
Got it! Is there a way to save my Modbus frames that will be sent to the slave as a variable in another place? I would like to save some script space.

**Reply by Erickrk** [reply]
Hey @gemu2015, pls let me know if you have any suggestions.

**Comment by gemu2015** [comment, marked as answer]
did you read the scripting docs ?
there is a section describing the options for larger script sizes.
on an esp8266 you may use about 6k max on an esp32 up to 16k
if you use the optional script editor you can preserve the most memory for the script
